evm_loader/program/src/account_storage/apply.rs

```diff
@@ -42,8 +43,9 @@ impl<'a> ProgramAccountStorage<'a> {
         let mut source = BalanceAccount::from_account(&crate::ID, source)?;
 
         let mut target = self.accounts.operator_balance();
-
-        source.increment_revision(&self.rent, &self.accounts)?;
+        if inc_revision {
+            source.increment_revision(&self.rent, &self.accounts)?;
```

Comment:
i think it can be skipped for non-iterative transactions, because the balance account has already been touched within the same transaction.

Adjusted trx_step and trx_execute, left a comment at the latter.

Reply:
Revisions must be incremented for all transactions. Regardless of their type.
If you changed the balance in any way, increment the revision.

With iterative transaction, balance is changed in the first and may be in the last iterations.
They doesn't reset on every iteration.

Reply:
My main point - we already currently increase the operator balance without incrementing its revision **on every iteration**:

`storage.consume_gas(used_gas, accounts.db().try_operator_balance())?;` in the `finalize`

Reply:
Operator uses the special accounts that does not participate in the transactions.

Reply:
You can't treat user's balance in the same way as operator's

Reply:
Got it. 

I reworked this whole case. Results:
- Removed inc_revision flag from `transfer_gas_payment`, it increments revision of the balance all the time (as it used to work).
- Introduced additional field in the StateAccount::Data named `priority_fee_used`, similar to gas_used with the same logic.
- Incorporated the usage of `priority_fee_used` in the `consume_gas` and `refund`, the same way it works for `gas_used`. 

P.S. The reason for new field - we can't really determine how much priority fee was spent solely from gas_used, because of the logic with CU limit and price. There's a logic that we discussed recently with Andrey (remove the validation & charge the `min(CU spent, max_priority_fee_per_gas)`) from the user. That's why we need a separate field to keep track of priority fee charged across iterations.